dlt/common/data_writers/writers.py

```diff
@@ -153,3 +156,74 @@ def write_footer(self) -> None:
     @classmethod
     def data_format(cls) -> TFileFormatSpec:
         return TFileFormatSpec("insert_values", "insert_values", False, False, requires_destination_capabilities=True)
+
+class ParquetDataWriter(DataWriter):
+
+    def __init__(self, f: IO[Any], caps: DestinationCapabilitiesContext = None) -> None:
+        super().__init__(f, caps)
+        from dlt.helpers.parquet_helper import pq
+        from dlt.helpers.parquet_helper import pyarrow
+
+        self.writer: Optional[pq.ParquetWriter] = None
+        self.schema: Optional[pyarrow.Schema] = None
+        self.complex_indices: List[str] = None
+
+    def write_header(self, columns_schema: TTableSchemaColumns) -> None:
+        from dlt.helpers.parquet_helper import pyarrow
+        from dlt.helpers.parquet_helper import pq
+
+        # build schema
+        self.schema = pyarrow.schema([pyarrow.field(name, self.get_data_type(schema_item["data_type"]), nullable=schema_item["nullable"]) for name, schema_item in columns_schema.items()])
+        # find row items that are of the complex type (could be abstracted out for use in other writers?)
+        self.complex_indices = [i for i, field in columns_schema.items() if field["data_type"] == "complex"]
+        self.writer = pq.ParquetWriter(self._f, self.schema, flavor="spark")
+
+
+    def write_data(self, rows: Sequence[Any]) -> None:
+        super().write_data(rows)
+        from dlt.helpers.parquet_helper import pyarrow
```

Comment:
the common code has many places like this. ie some credentials lazy import bigquery or boto3. I could not find any good pattern to do that. the guiding principle is that dlt can be used with minimum (default) credentials and it asks for extras via this specialized exceptions

Reply:
I have moved the parquet stuff now into a newly created libs folder, maybe this would be a nice pattern for the wrappers of external libraries, not sure, lmk